File: slash_commands/poll.py
#!/usr/bin/env python3
import discord


async def poll(ctx, question, option1, option2, option3=None, option4=None, option5=None, option6=None, option7=None, option8=None, option9=None):

    embed = discord.Embed(color=0xfee9b6, title="⏳  Loading...")
    interaction = await ctx.respond(embed=embed)

    options = [option1, option2, option3, option4, option5, option6, option7, option8, option9]
    numbers = ["1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️⃣", "7️⃣", "8️⃣", "9️⃣"]
    description = ""
    optionCount = 0

    for option in options:
        if option:
            description += "\n" + numbers[optionCount] + ".  " + option
            optionCount += 1

    embed = discord.Embed(color=0xffd983,
                          title=f"🗳️  Poll:  {question}",
                          description=description
                          )

    message = await interaction.edit_original_response(embed=embed)

    for count in range(optionCount):
        await message.add_reaction(numbers[count])

File: commands/poll.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
  ButtonStyle,
} from "../types/discord";
import { embedResponse, createButtonRow } from "../utils/discord";
import { getOptionValue } from "../handlers/commands";

const NUMBER_EMOJIS = ["1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️⃣", "7️⃣", "8️⃣", "9️⃣"];

export async function handlePoll(
  interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const question = getOptionValue<string>(interaction, "question")!;
  const options: (string | undefined)[] = [
    getOptionValue<string>(interaction, "option1"),
    getOptionValue<string>(interaction, "option2"),
    getOptionValue<string>(interaction, "option3"),
    getOptionValue<string>(interaction, "option4"),
    getOptionValue<string>(interaction, "option5"),
    getOptionValue<string>(interaction, "option6"),
    getOptionValue<string>(interaction, "option7"),
    getOptionValue<string>(interaction, "option8"),
    getOptionValue<string>(interaction, "option9"),
  ];

  // Filter out undefined options
  const validOptions = options.filter(
    (opt): opt is string => opt !== undefined,
  );

  // Build the description with numbered options
  let description = "";
  validOptions.forEach((option, index) => {
    description += `\n${NUMBER_EMOJIS[index]}  ${option}`;
  });

  // Create button rows for voting (max 5 buttons per row)
  const components: ReturnType<typeof createButtonRow>[] = [];

  // First row (up to 5 options)
  const firstRowButtons = validOptions.slice(0, 5).map((_, index) => ({
    customId: `poll_vote_${index}`,
    label: NUMBER_EMOJIS[index],
    style: ButtonStyle.SECONDARY,
  }));
  if (firstRowButtons.length > 0) {
    components.push(createButtonRow(firstRowButtons));
  }

  // Second row (options 6-9)
  const secondRowButtons = validOptions.slice(5, 9).map((_, index) => ({
    customId: `poll_vote_${index + 5}`,
    label: NUMBER_EMOJIS[index + 5],
    style: ButtonStyle.SECONDARY,
  }));
  if (secondRowButtons.length > 0) {
    components.push(createButtonRow(secondRowButtons));
  }

  return {
    type: 4,
    data: {
      embeds: [
        {
          title: `🗳️ Poll: ${question}`,
          description: description + "\n\n*Click a button to vote!*",
          color: Colors.YELLOW,
        },
      ],
      components,
    },
  };
}
